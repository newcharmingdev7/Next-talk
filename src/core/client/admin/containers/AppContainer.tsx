import React from "react";

import { AppContainerQueryResponse } from "coral-admin/__generated__/AppContainerQuery.graphql";

import { graphql } from "coral-framework/lib/relay";
import { withRouteConfig } from "coral-framework/lib/router";

import App from "../components/App";

interface Props {
  data: AppContainerQueryResponse | null;
}

class AppContainer extends React.Component<Props> {
  public render() {
    return (
      <App viewer={this.props.data && this.props.data.viewer}>
        {this.props.children}
      </App>
    );
  }
}

const enhanced = withRouteConfig<Props>({
  query: graphql`
    query AppContainerQuery {
      viewer {
        ...UserMenuContainer_viewer
        ...NavigationContainer_viewer
      }
    }
  `,
})(AppContainer);

export default enhanced;
