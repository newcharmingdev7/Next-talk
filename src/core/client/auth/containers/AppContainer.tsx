import * as React from "react";
import { StatelessComponent } from "react";

import App from "../components/App";

const AppContainer: StatelessComponent = () => {
  return <App />;
};

export default AppContainer;
