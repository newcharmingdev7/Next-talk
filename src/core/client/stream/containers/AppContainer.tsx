import React from "react";
import { withFragmentContainer } from "talk-framework/lib/relay";
import { graphql, withLocalStateContainer } from "talk-framework/lib/relay";
import { AppContainerLocal as Local } from "talk-stream/__generated__/AppContainerLocal.graphql";
import {
  SetActiveTabInput,
  SetActiveTabMutation,
  withSetActiveTabMutation,
} from "talk-stream/mutations";

import { AppContainer_asset as AssetData } from "talk-stream/__generated__/AppContainer_asset.graphql";
import App from "../components/App";

interface InnerProps {
  local: Local;
  setActiveTab: SetActiveTabMutation;
  asset: AssetData;
}

class AppContainer extends React.Component<InnerProps> {
  private handleSetActiveTab = (tab: SetActiveTabInput["tab"]) => {
    this.props.setActiveTab({ tab });
  };

  public render() {
    const {
      local: { activeTab },
      asset,
    } = this.props;

    return (
      <App
        activeTab={activeTab}
        onTabClick={this.handleSetActiveTab}
        commentCount={asset.commentCounts.totalVisible}
      />
    );
  }
}

const enhanced = withSetActiveTabMutation(
  withFragmentContainer<InnerProps>({
    asset: graphql`
      fragment AppContainer_asset on Asset {
        commentCounts {
          totalVisible
        }
      }
    `,
  })(
    withLocalStateContainer(
      graphql`
        fragment AppContainerLocal on Local {
          activeTab
        }
      `
    )(AppContainer)
  )
);

export default enhanced;
