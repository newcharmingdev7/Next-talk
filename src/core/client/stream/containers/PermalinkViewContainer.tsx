import React, { StatelessComponent } from "react";
import { graphql } from "react-relay";
import { withFragmentContainer } from "talk-framework/lib/relay";
import { withLocalStateContainer } from "talk-framework/lib/relay";
import { PropTypesOf } from "talk-framework/types";
import { AppQueryLocal as Local } from "talk-stream/__generated__/AppQueryLocal.graphql";
import { PermalinkViewContainerQuery as Data } from "talk-stream/__generated__/PermalinkViewContainerQuery.graphql";
import PermalinkView from "../components/Permalink/PermalinkView";

interface InnerProps {
  data: Data;
  local: Local;
}

export const PermalinkViewContainer: StatelessComponent<InnerProps> = ({
  local,
  data,
}) => {
  return <PermalinkView {...data} assetURL={local.assetURL} />;
};

const enhanced = withFragmentContainer<{ data: Data }>({
  data: graphql`
    fragment PermalinkViewContainerQuery on Query
      @argumentDefinitions(commentID: { type: "ID!" }) {
      comment(id: $commentID) {
        ...CommentContainer
      }
    }
  `,
})(
  withLocalStateContainer<Local>(
    graphql`
      fragment PermalinkViewContainerLocal on Local {
        assetURL
      }
    `
  )(PermalinkViewContainer)
);

export type PermalinkViewContainerProps = PropTypesOf<typeof enhanced>;
export default enhanced;
