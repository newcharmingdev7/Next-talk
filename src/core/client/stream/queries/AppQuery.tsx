import * as React from "react";
import { StatelessComponent } from "react";
import { ReadyState } from "react-relay";

import {
  graphql,
  QueryRenderer,
  withLocalStateContainer,
} from "talk-framework/lib/relay";
import {
  AppQueryResponse,
  AppQueryVariables,
} from "talk-stream/__generated__/AppQuery.graphql";
import { AppQueryLocal as Local } from "talk-stream/__generated__/AppQueryLocal.graphql";

import PermalinkViewContainer from "talk-stream/containers/PermalinkViewContainer";
import AppContainer from "../containers/AppContainer";

interface InnerProps {
  local: Local;
}

interface WrappedProps {
  data: any;
}

// TODO (bc) Break down the needs of each component into another file (maybe).
// (careful porting QueryRenderer into another stateless component)

export const renderWrapper = (
  WrappedComponent: React.ComponentType<WrappedProps>
) => ({ error, props }: ReadyState<AppQueryResponse>) => {
  if (error) {
    return <div>{error.message}</div>;
  }
  if (props) {
    return <WrappedComponent data={props} />;
  }
  return <div>Loading</div>;
};

const AppQuery: StatelessComponent<InnerProps> = ({
  local: { commentID, assetID },
}) => {
  if (commentID) {
    return (
      <QueryRenderer<AppQueryVariables, AppQueryResponse>
        query={graphql`
          query AppQuery_PermalinkViewContainer_Query($commentID: ID!) {
            ...PermalinkViewContainerQuery @arguments(commentID: $commentID)
          }
        `}
        variables={{
          commentID,
        }}
        render={renderWrapper(PermalinkViewContainer)}
      />
    );
  }

  return (
    <QueryRenderer<AppQueryVariables, AppQueryResponse>
      query={graphql`
        query AppQuery($assetID: ID!) {
          ...AppContainer @arguments(assetID: $assetID)
        }
      `}
      variables={{
        assetID,
      }}
      render={renderWrapper(AppContainer)}
    />
  );
};

const enhanced = withLocalStateContainer<Local>(
  graphql`
    fragment AppQueryLocal on Local {
      assetID
      commentID
      assetURL
    }
  `
)(AppQuery);

export default enhanced;
