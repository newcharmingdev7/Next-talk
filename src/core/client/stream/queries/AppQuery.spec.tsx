import { shallow } from "enzyme";
import React from "react";

import PermalinkViewContainer from "talk-stream/containers/PermalinkViewContainer";
import AppContainer from "../containers/AppContainer";
import { renderWrapper } from "./AppQuery";

it("renders permalink view", () => {
  const data = {
    props: {} as any,
    error: null,
  };
  const component = renderWrapper(PermalinkViewContainer);
  const wrapper = shallow(React.createElement(() => component(data)));
  expect(wrapper).toMatchSnapshot();
});

it("renders app", () => {
  const data = {
    props: {} as any,
    error: null,
  };
  const component = renderWrapper(AppContainer);
  const wrapper = shallow(React.createElement(() => component(data)));
  expect(wrapper).toMatchSnapshot();
});

it("renders loading", () => {
  const data = {
    props: null,
    error: null,
  };
  const component = renderWrapper(AppContainer);
  const wrapper = shallow(React.createElement(() => component(data)));
  expect(wrapper).toMatchSnapshot();
});

it("renders error", () => {
  const data = {
    props: null,
    error: new Error("error"),
  };
  const component = renderWrapper(AppContainer);
  const wrapper = shallow(React.createElement(() => component(data)));
  expect(wrapper).toMatchSnapshot();
});
