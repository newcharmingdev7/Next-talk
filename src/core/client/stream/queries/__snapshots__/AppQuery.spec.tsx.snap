// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders app 1`] = `
<Relay(AppContainer)
  data={Object {}}
/>
`;

exports[`renders error 1`] = `
<div>
  error
</div>
`;

exports[`renders loading 1`] = `
<div>
  Loading
</div>
`;

exports[`renders permalink view 1`] = `
<Relay(PermalinkViewContainer)
  data={Object {}}
/>
`;
